src/outmdsd/DjsonLogItem.cc

```diff
@@ -47,32 +47,16 @@ DjsonLogItem::ComposeSchema(
 {
     IdMgr::value_type_t cachedInfo;
 
-    // because most log items will share same schema, use unsorted key first
-    auto unsortedKey = GetSchemaCacheKey();
-    if (GetIdMgr().GetItem(unsortedKey, cachedInfo)) {
+    // The fields order are preserved. So schema with same names/types
+    // but in different order will be treated different schemas.
+    auto key = GetSchemaCacheKey();
+    if (GetIdMgr().GetItem(key, cachedInfo)) {
         strm << cachedInfo.first << "," << cachedInfo.second;
```

Comment:
If you found the key in the cache, then you've already sent the schema to mdsd at least once on this connection, and you don't have to send it again. The mdsd code (```ProtocolHandlerJSON.cc::decodeMsg()```) is just fine with seeing a null here. So this line should be ```strm << cachedInfo.first << ",null";```